clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -121,8 +122,9 @@ public LeaderAndIsrRequest(Struct struct, short version) {
             List<Integer> replicas = new ArrayList<>(replicasArray.length);
             for (Object r : replicasArray)
                 replicas.add((Integer) r);
+            boolean isNew = partitionStateData.hasField(IS_NEW_KEY_NAME) ? partitionStateData.getBoolean(IS_NEW_KEY_NAME) : false;
```

Comment:
Regarding the first comment, it seems safer to disable replica creation if any replica offline. I think it is probably more backward compatible to do so -- currently LeaderAndIsrRequest will fail to create replica if any disk is offline.

Regarding the second comment, good point. I should have updated the controller code to specify the version for LeaderAndIsrRequest and UpdateMetadataRequest based on the configured `interBrokerProtocolVersion`. I have updated the patch to do this and added `KAFKA_0_11_1_IV0` in `ApiVersion`. Thanks!